File: packages/shared/src/worker.ts
/** Mirrors flowforge::domain::WorkerStatus (engine/include/flowforge/domain/worker.hpp). */
export type WorkerStatus = "idle" | "busy" | "offline";

export interface Worker {
  id: string;
  hostname: string;
  status: WorkerStatus;
  registered_at: string;
  last_heartbeat: string;
}

export interface ListWorkersResponse {
  workers: Worker[];
}

File: engine/src/domain/worker.cpp
#include "flowforge/domain/worker.hpp"

namespace flowforge::domain {

std::string_view to_string(WorkerStatus status) noexcept {
  switch (status) {
    case WorkerStatus::Idle:
      return "idle";
    case WorkerStatus::Busy:
      return "busy";
    case WorkerStatus::Offline:
      return "offline";
  }
  return "unknown";
}

}  // namespace flowforge::domain
